operator/controllers/resources/role_controller_test.go

```diff
@@ -73,6 +75,62 @@ func TestRoleCreation(t *testing.T) {
 	})
 }
 
+func TestRoleCreationFromYAML(t *testing.T) {
+	ctx := context.Background()
+
+	teleportServer, operatorName := defaultTeleportServiceConfig(t)
+
+	require.NoError(t, teleportServer.Start())
+
+	tClient := clientForTeleport(t, teleportServer, operatorName)
+	k8sClient := startKubernetesOperator(t, tClient)
+
+	namespace := createNamespaceForTest(t, k8sClient)
+	roleName := "interns"
+
+	roleYAML := yaml.NewYAMLOrJSONDecoder(bytes.NewBuffer([]byte(`
+kind: role
+version: v5
+metadata:
+  name: interns
+spec:
+  allow:
+    node_labels:
+      '*': ['*']
+`)), 1024)
+
+	var roleX resourcesv5.TeleportRole
+	err := roleYAML.Decode(&roleX)
+	require.NoError(t, err)
```

Comment:
Where does the test break before the change? Could we add some messages to the require calls so it's easier to distinguish what is setup and what is the focus of the test?

Reply:
I refacto-ed the test, is this clearer now ?

Reply:
Better, thanks!